basesandbox/src/ipc/mod.rs

```diff
@@ -0,0 +1,28 @@
+// Copyright 2020 Kodebox, Inc.
+// This file is part of CodeChain.
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as
+// published by the Free Software Foundation, either version 3 of the
+// License, or (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+
+pub mod domain_socket;
+pub mod semaphore;
+
+pub trait Ipc: Drop {
+    /// This distinction may be required for some asymmetry IPC schemes
```

Comment:
Do we have a need for an asymmetry IPC?
How about distinguishing a server and a client when we really need an asymmetric IPC?

Reply:
Ah there was a mistake. `executee::start()` should call `T::create_client`. 
For example socket stream (not datagram) will differently implement those two.

Reply:
Do you have a plan to support the socket stream?

Reply:
I'm not sure. If you think such distinction is unnecessary, then I'll just remove it.